posthog-core/src/gzip.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Older browsers and some runtimes don't support this yet
+ */
+export function isGzipSupported(): boolean {
+  return 'CompressionStream' in globalThis
+}
+
+/**
+ * Gzip a string using Compression Streams API if it's available
+ */
+export async function gzipCompress(input: string, isDebug = true): Promise<Blob | null> {
+  if (!isGzipSupported()) return null
+  try {
+    // Turn the string into a stream using a Blob, and then compress it
+    const dataStream = new Blob([input], {
+      type: 'text/plain',
+    }).stream()
+
+    const compressedStream = dataStream.pipeThrough(new CompressionStream('gzip'))
+
+    // Using a Response to easily extract the readablestream value. Decoding into a string for fetch
+    return await new Response(compressedStream).blob()
+  } catch (error) {
+    if (isDebug) console.error('Failed to gzip compress data', error)
```

Comment:
you can use `logMsgIfDebug` instead

Reply:
true its a different file, lets keep this for now then.
regarding being cleaner, I agree but its much easier to refactor once things change since we use the same function across different packages, etc.